app/lib/features/onboarding/actions/generate_invitecode_externally_actions.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/features/super_invites/providers/super_invites_providers.dart';
+import 'package:acter/features/super_invites/utils.dart';
+import 'package:acter/l10n/generated/l10n.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_easyloading/flutter_easyloading.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::invite::generate_invitecode_externally_actions');
+
+Future<String> generateNewInviteCode(
+    BuildContext context,
+    WidgetRef ref,
+    String roomId,
+  ) async {
+    final lang = L10n.of(context);
+    try {
+      EasyLoading.show(status: lang.generateInviteCode);
+      final displayName = await ref.read(
+        roomDisplayNameProvider(roomId).future,
+      );
+      final inviteCode = generateInviteCodeName(displayName);
+
+      await newSuperInviteForRooms(ref, [
+        roomId,
+      ], inviteCode: inviteCode);
+      ref.invalidate(superInvitesProvider);
+      EasyLoading.dismiss();
+      return inviteCode;
```

Comment:
```suggestion
      final generatedInviteCode = await newSuperInviteForRooms(ref, [
        roomId,
      ], inviteCode: inviteCode);
      ref.invalidate(superInvitesProvider);
      EasyLoading.dismiss();
      return generatedInviteCode;
```

`newSuperInviteForRooms` is the main function which is actually generating invite code and returning it.